packages/seal/src/client.ts

```diff
@@ -312,27 +346,29 @@ export class SealClient {
 				// Check if all the receivedIds are consistent with the requested fullIds.
 				// If so, consider the key server got all keys and mark as completed.
 				if (fullIds.every((fullIds) => receivedIds.has(fullIds))) {
-					completedServerCount++;
+					completedWeight += this.#weight(objectId)!;
 
 					// Return early if the completed servers is more than the threshold.
-					if (completedServerCount >= threshold) {
+					if (completedWeight >= threshold) {
 						controller.abort();
 					}
 				}
 			} catch (error) {
 				if (!controller.signal.aborted) {
 					errors.push(error as Error);
 				}
+			} finally {
 				// If there are too many errors that the threshold is not attainable, return early with error.
-				if (remainingKeyServers.size - errors.length < threshold - completedServerCount) {
-					controller.abort(error);
+				remainingKeyServersWeight -= this.#weight(objectId)!;
+				if (remainingKeyServersWeight < threshold - completedWeight) {
+					controller.abort(new TooManyFailedFetchKeyRequestsError());
 				}
 			}
 		});
 
 		await Promise.allSettled(keyFetches);
 
-		if (completedServerCount < threshold) {
+		if (completedWeight < threshold) {
```

Comment:
should we change `toMajorityError(errors)` to ignore `TooManyFailedFetchKeyRequestsError`?
say we have 3 servers and threshold 3, and we got NoAccess from one of them -> `TooManyFailedFetchKeyRequestsError` will appear twice and be returned by `toMajorityError`, but the important error here is NoAccess

Reply:
I think this is handled already since an error is only added to the list of errors if it's not from an abort.